src/components/ProjectList/index.jsx

```diff
@@ -0,0 +1,112 @@
+import { useState, useEffect } from 'react';
+import cls from 'classnames';
+import TabToggle from '../TabToggle';
+import { colors, flex, layout, margin, typography as type } from '../../theme';
+import store from '../../store';
+import { getProjectsArchive } from '../../actions';
+import ProjectListItem from '../ProjectListItem';
+
+export default function ProjectList({
+  selectedProject,
+  setSelectedProject,
+  isArchive = false,
+  setIsArchive,
+}) {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [projectsList, setProjectsList] = useState([]);
+  const [archiveList, setArchivesList] = useState([]);
+  const [selectedList, setSelectedList] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      if (selectedIndex === 0) {
+        const projects = await store.getProjectsList();
+        setProjectsList(projects);
+        setSelectedProject(projects[0]);
+      } else {
+        const archive = await getProjectsArchive();
+        setArchivesList(archive);
+        setSelectedProject(archive[0]);
+      }
+      setIsArchive(selectedIndex === 1);
+    })();
+  }, [selectedIndex]);
```

Comment:
This is the start of the "two highway" solution for managing the project list view. The only main issue is the data shape of the two lists are slightly different. ProjectsList is just an array of strings of the slugs while archiveList is an array of objects that contains both the slug and full name for display in the toolbar.

Reply:
Actually, I have a few thoughts about this file that would be more complicated to type out. When you get a chance to go through this review, let's set up a quick chat to discuss this. I feel like there's some duplicated state here that's being used that might not be necessary.